utils/build-script

```diff
@@ -322,87 +322,98 @@ class BuildScriptInvocation(object):
         # FIXME: We should move the platform-derived arguments to be entirely
         # data driven, so that we can eliminate this code duplication and just
         # iterate over all supported platforms.
+        self.platforms_to_skip_build = self.__platforms_to_skip_build(args)
+        self.platforms_to_skip_test = self.__platforms_to_skip_test(args)
+        self.platforms_archs_to_skip_test = \
+            self.__platforms_archs_to_skip_test(args)
+        self.platforms_to_skip_test_host = \
+            self.__platforms_to_skip_test_host(args)
 
-        self.platforms_to_skip_build = set()
+        self.build_libparser_only = args.build_libparser_only
+
+    def __platforms_to_skip_build(self, args):
```

Comment:
Nitpick: I think the Python standard for "private" methods is to prefix with a single `_` character.

Reply:
One underscore is a “agreement” of a non-public API, but completely accessible. At least two underscores (and not finishing in more than one underscore) will ask the Python interpreter to mangle the name, becoming a little bit more “private” (but the mangling scheme is quite simple, so still accessible).

(Ref: https://docs.python.org/2.7/tutorial/classes.html#private-variables-and-class-local-references)

In this case in particular, it doesn’t matter that much which one to use (anything that indicates “don’t call this” should be fine), but in other cases (specially when inheritance is involved), the two underscores avoids name clashing problems.

Reply:
I think I remember seeing something along these lines a long time ago, but I totally forgot. That's a pretty cool feature! I'm totally fine with keeping these names then considering these are implementation details.